### PR TITLE
feat(cloudtable): add new cluster resource support

### LINKED ISSUE #1948: [CLoudTable] Needs a new cluster resource for terraform to use
Hi team,

 The old resource `huaweicloud_cloudtable_cluster` cannot be used normally, and a new resource are required to support the use of terraform scripts.

Best regards!

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
At present, the old resource can no longer be used normally, and a new cluster resource is designed and launched according to the HuaweiCloud API.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #1948 

**Special notes for your reviewer**:

**Release note**:

```release-note
1. refactor cluster resource of the CloudTable service.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/cloudtable' TESTARGS='-run=TestAccCloudtableCluster_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cloudtable -v -run=TestAccCloudtableCluster_basic -timeout 360m -parallel 4
=== RUN   TestAccCloudtableCluster_basic
=== PAUSE TestAccCloudtableCluster_basic
=== CONT  TestAccCloudtableCluster_basic
--- PASS: TestAccCloudtableCluster_basic (666.46s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cloudtable     666.531s
```

### CODE REVIEW

File: huaweicloud/services/cloudtable/resource_huaweicloud_cloudtable_cluster.go [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+	"300": "Abnormal",
+	"303": "Creation failed",
+	"800": "Frezon",
+}
+
+func ResourceCloudTableCluster() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceCloudTableClusterV2Create,
+		ReadContext:   resourceCloudTableClusterV2Read,
+		DeleteContext: resourceCloudTableClusterV2Delete,
```
Comment (ShiChangkuo) [inline huaweicloud/services/cloudtable/resource_huaweicloud_cloudtable_cluster.go]: can the resource be imported?

File: docs/resources/cloudtable_cluster.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  lowercase letters, numbers and hyphens (-). Changing this parameter will create a new resource.
+
+* `storage_type` - (Required, String, ForceNew) Specifies the storage type.
+  The valid values are **COMMON** and **ULTRAHIGH**. Changing this parameter will create a new resource.
+
+* `vpc_id` - (Required, String, ForceNew) Specifies the VPC ID to which the cluster belongs.
+  Changing this parameter will create a new resource.
+
+* `network_id` - (Required, String, ForceNew) Specifies the ID of the network to which the cluster belongs.
+  Changing this parameter will create a new resource.
```
Comment (hmbig2) [inline docs/resources/cloudtable_cluster.md:57]: is it subnet_id?

File: huaweicloud/services/cloudtable/resource_huaweicloud_cloudtable_cluster.go [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+func resourceCloudTableClusterV2Delete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	client, err := config.CloudtableV2Client(config.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("Error creating HuaweiCloud CloudTable v2 client: %s", err)
+	}
+
+	err = clusters.Delete(client, d.Id()).ExtractErr()
+	if err != nil {
+		return fmtp.DiagErrorf("Error deleting the specifies namespace (%s): %s", d.Id(), err)
```
Comment (hmbig2) [inline huaweicloud/services/cloudtable/resource_huaweicloud_cloudtable_cluster.go]: s/namespace/CloudTable cluster

File: huaweicloud/services/cloudtable/resource_huaweicloud_cloudtable_cluster.go [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+		Refresh:        clusterStateRefreshFunc(c, clusterId),
+		Timeout:        s.Timeout,
+		Delay:          s.Delay,
+		PollInterval:   s.PollInterval,
+		NotFoundChecks: s.NotFoundChecks,
+	}
+	_, err := stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return fmtp.DiagErrorf("Waiting for the status of the namespace (%s) to complete (%s) timeout: %s",
+			clusterId, s.Target, err)
```
Comment (hmbig2) [inline huaweicloud/services/cloudtable/resource_huaweicloud_cloudtable_cluster.go:316]: ditto

File: huaweicloud/services/cloudtable/resource_huaweicloud_cloudtable_cluster.go [hunk 5]
```diff
@@ -0,0 +1,347 @@
...
+	client, err := config.CloudtableV2Client(config.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("Error creating HuaweiCloud CloudTable v2 client: %s", err)
+	}
+
+	resp, err := clusters.Get(client, d.Id())
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "Error getting the specifies CloudTable cluster form server")
+	}
+	logp.Printf("The type of the parameter storage quota is %v", reflect.TypeOf(resp.StorageQuota))
```
Comment (hmbig2) [inline huaweicloud/services/cloudtable/resource_huaweicloud_cloudtable_cluster.go]: miss the log level

Comment (Lance52259) [reply huaweicloud/services/cloudtable/resource_huaweicloud_cloudtable_cluster.go]: This print msg was used for testing, now removed.

File: docs/resources/cloudtable_cluster.md (hunk 2 above)
Comment (Lance52259) [reply docs/resources/cloudtable_cluster.md:57]: `network_id` is used to ensure compatibility.
There is a problem with the variable reference in the sample.
```
network_id = var.subnet_id
```
should be network id (ID of the resource `huaweicloud_vpc_subnet`).

File: huaweicloud/services/cloudtable/resource_huaweicloud_cloudtable_cluster.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/services/cloudtable/resource_huaweicloud_cloudtable_cluster.go]: supported

File: docs/resources/cloudtable_cluster.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+* `opentsdb_public_endpoint` - The OpenTSDB public network endpoint address.
+
+* `storage_size` - The storage size, in GB.
+
+* `used_storage_size` - The currently used storage, in GB.
+
+* `zookeeper_link` - The intranet zookeeper connection access address.
+
+* `subnet_id` - The subnet ID to which the cluster belongs.
```
Comment (ShiChangkuo) [inline docs/resources/cloudtable_cluster.md]: suggest to delete this attribute

File: huaweicloud/services/cloudtable/resource_huaweicloud_cloudtable_cluster.go [hunk 7]
```diff
@@ -0,0 +1,366 @@
...
+
+func buildCloudTableClusterParams(d *schema.ResourceData) clusters.CreateOpts {
+	result := clusters.CreateOpts{
+		Name:           d.Get("name").(string),
+		VpcId:          d.Get("vpc_id").(string),
+		StorageType:    d.Get("storage_type").(string),
+		IAMAuthEnabled: d.Get("iam_auth_enabled").(bool),
+		Datastore: clusters.Datastore{
+			Type:    "hbase",
+			Version: "1.0.6",
```
Comment (ShiChangkuo) [inline huaweicloud/services/cloudtable/resource_huaweicloud_cloudtable_cluster.go]: the latest version is 1.3.1, please update it. Also we can add an attribute to show the hbase version.

File: huaweicloud/services/cloudtable/resource_huaweicloud_cloudtable_cluster.go [hunk 8]
```diff
@@ -0,0 +1,366 @@
...
+func setCloudTableClusterStatus(d *schema.ResourceData, strCode string) error {
+	if val, ok := stateCodes[strCode]; ok {
+		return d.Set("status", val)
+	}
+	return fmtp.Errorf("the status is abnormal, the code is %+v")
+}
+
+func setCloudTableClusterTSDNum(d *schema.ResourceData, strNum string) error {
+	tsdNum, err := strconv.Atoi(strNum)
+	if err != nil {
```
Comment (ShiChangkuo) [inline huaweicloud/services/cloudtable/resource_huaweicloud_cloudtable_cluster.go]: do not raise error when converting failed, just set the field when err is nil

File: huaweicloud/services/cloudtable/resource_huaweicloud_cloudtable_cluster.go [hunk 9]
```diff
@@ -0,0 +1,366 @@
...
+		return err
+	}
+	return d.Set("opentsdb_num", tsdNum)
+}
+
+func setCloudTableClusterRSNum(d *schema.ResourceData, strNum string) error {
+	tsdNum, err := strconv.Atoi(strNum)
+	if err != nil {
```
Comment (ShiChangkuo) [inline huaweicloud/services/cloudtable/resource_huaweicloud_cloudtable_cluster.go]: ditto

File: huaweicloud/services/cloudtable/resource_huaweicloud_cloudtable_cluster.go (hunk 7 above)
Comment (Lance52259) [reply huaweicloud/services/cloudtable/resource_huaweicloud_cloudtable_cluster.go]: The parameter `hbase_version` has been supported at **3e33e71**.

File: huaweicloud/services/cloudtable/resource_huaweicloud_cloudtable_cluster.go (hunk 8 above)
Comment (Lance52259) [reply huaweicloud/services/cloudtable/resource_huaweicloud_cloudtable_cluster.go]: done

File: docs/resources/cloudtable_cluster.md [hunk 10]
```diff
@@ -0,0 +1,121 @@
+---
+subcategory: "Cloud Table"
```
Comment (ShiChangkuo) [inline docs/resources/cloudtable_cluster.md]: please use **CloudTable**

File: docs/resources/cloudtable_cluster.md (hunk 6 above)
Comment (Lance52259) [reply docs/resources/cloudtable_cluster.md]: done

**Pull request merged**